asset/asset.go

```diff
@@ -462,6 +464,66 @@ const (
 	ScriptV0 ScriptVersion = 0
 )
 
+// TapscriptTreePreimage represents the two supported ways to define a tapscript
+// tree to be used as a sibling for a Taproot Asset commitment, an asset group
+// key, or an asset script key.
+type TapscriptTreePreimage = fn.Either[TapBranchPreimage, []txscript.TapLeaf]
+
+func NewTreePreimageFromBranchPreimage(
+	b TapBranchPreimage) TapscriptTreePreimage {
+
+	return fn.NewLeft[TapBranchPreimage, []txscript.TapLeaf](b)
+}
+
+func NewTreePreimageFromLeaves(l []txscript.TapLeaf) (*TapscriptTreePreimage,
+	error) {
+
+	if len(l) == 0 {
+		return nil, fmt.Errorf("no leaves given")
+	}
+	// TODO(jhb): assert leaf versions?
```

Comment:
Can assert that it's the default leaf version. Based on some recent investigation, we'll sorta  move away from having any real requirements re the leaf version at all.